parachain-template/node/src/cli.rs

```diff
@@ -75,12 +75,9 @@ impl RelayChainCli {
 	) -> Self {
 		let extension = crate::chain_spec::Extensions::try_get(&*para_config.chain_spec);
 		let chain_id = extension.map(|e| e.relay_chain.clone());
-		let base_path = para_config
-			.base_path
-			.as_ref()
-			.map(|x| x.path().join("polkadot"));
+		let base_path = para_config.base_path.path().join("polkadot");
 		Self {
-			base_path,
+			base_path: Some(base_path),
```

Comment:
Instead of adding this Some, you can change the definition from

    pub base_path: Option<PathBuf>,

To

    pub base_path: PathBuf,

above

Reply:
I will not be possible to apply this change, because the `CliConfiguration` trait present in `substrate/client/cli/src/config.rs` has a `base_path()` method that returns a result containing an `Option<BasePath>` and conflicts if we change the definition of our `base_path` field.

Reply:
Can't you just add a `Some` to that method that needs it?

Reply:
Like this:

```rust
fn base_path(&self) -> Result<Option<BasePath>> {
        Ok(self
            .shared_params()
            .base_path()?
            .or_else(|| Some(self.base_path.clone().into())))
}
```